Combi2k2/feature/message schema

Combi2k2 opened the pull request:
Message Schema:
- Serialize
- Deserialize

Combi2k2 committed:
feat: define message schema + methods

Combi2k2 committed:
test: add test for message method

Combi2k2 committed:
fix: build error on local machine

xmen1404 commented on core/base/message.h:
```diff
@@ -1,17 +1,70 @@
+    }
 };
 
 struct Message {
```
why need pointer?

xmen1404 commented on core/base/message.h:
```diff
@@ -1,17 +1,70 @@
+    MessageHeader *header;
+    MessageBody *body;
+
+    Message(MessageType t, uint64_t src, uint64_t dst, const std::string &payload, time_t timestamp) {
```
should use `std::span<char>` instead of const `std::string& payload`

xmen1404 commented on core/base/message.h:
```diff
@@ -1,17 +1,70 @@
+    uint64_t dst;
+    size_t length;
+    time_t timestamp;
 
```
add default ctor: MessageHeader() {}

Combi2k2 committed:
fix:
- use instance instead of pointer in struct definition
- use std::span in Message constructor

xmen1404 commented on core/base/message.h:
```diff
@@ -1,17 +1,80 @@
+    ~MessageBody() {
+        delete[] buff;
+    }
+    void init(std::span<char> payload) {
```
`void Init(...)`

xmen1404 commented on core/base/message.h:
```diff
@@ -1,17 +1,80 @@
+        body.init(payload);
+    }
+
+    void serialize(char *buffer, size_t& offset) {
```
`size_t Serialize(char *buffer) {}`

xmen1404 replied:
/**
 * Serialize Message content to raw binary buffer
 * param: buffer pointer (char*)
 * return: message size (size_t)
 */

xmen1404 commented on core/base/message.h:
```diff
@@ -1,17 +1,80 @@
+            offset += header.length;
+        }
+    }
+    void deserialize(const char *buffer, size_t& offset) {
```
`size_t Deserialize(const char *buffer) {}`

Combi2k2 committed:
fix: fix style of naming class methods + fix type of Serialize + Deserialize methods
